### ISSUE TITLE
Please help to setup the YPTSocket

### ISSUE THREAD

**Issue by peterflitz** [WWBN/AVideo#5656]
YPTSocket will not run:

> root@tube:~# php /var/www/html/tube/plugin/YPTSocket/test.php
[2021-09-06 17:01:03] ** Starting socket test **
[2021-09-06 17:01:03] Testing connection with [0]: ws://localhost:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03] Testing connection with [1]: wss://localhost:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03] Testing connection with [2]: ws://127.0.0.1:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03] Testing connection with [3]: wss://127.0.0.1:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03] Testing connection with [4]: ws://tube.domain.com:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03] Testing connection with [5]: wss://tube.domain.com:2051 on 2021/09/06 17:01:03
[2021-09-06 17:01:03]  NOT SECURE CONNECTION  Good news, your localhost connects, that means your port [2051] is open ws://localhost:2051
[2021-09-06 17:01:23]    SECURE CONNECTION    CONNECTION FAIL wss://localhost:2051 Connection to tls://localhost:2051 timed out after 20 seconds
[2021-09-06 17:01:23]  NOT SECURE CONNECTION  CONNECTION FAIL ws://127.0.0.1:2051 Connection to tcp://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 17:01:23]    SECURE CONNECTION    CONNECTION FAIL wss://127.0.0.1:2051 Connection to tls://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 17:01:23]  NOT SECURE CONNECTION  CONNECTION FAIL ws://tube.domain.com:2051 Connection to tcp://tube.domain.com:2051 timed out after 20 seconds
[2021-09-06 17:01:23]    SECURE CONNECTION    CONNECTION FAIL wss://tube.domain.com:2051 Connection to tls://tube.domain.com:2051 timed out after 20 seconds
[2021-09-06 17:01:23]  We found 0 possible configurations:

**Comment by DanielnetoDotCom** [comment]
Make sure you have the server running in another terminal before you run the test

open another terminal and run `php /var/www/html/tube/plugin/YPTSocket/server.php`

only after that run the test

**Comment by peterflitz** [comment]
hmmh, not secure .
in the settings is the same certificat like from the website, the website is secure (https)

> Terminal 1:
root@tube:~# php /var/www/html/tube/plugin/YPTSocket/server.php
Starting AVideo Socket server version 2.6 on port 2051
Your socket server does NOT use a secure connection
2021-09-06 17:51:24 Construct


Terminal 2.
root@tube:~# php /var/www/html/tube/plugin/YPTSocket/test.php
[2021-09-06 17:51:29] ** Starting socket test **
[2021-09-06 17:51:29] Testing connection with [0]: ws://localhost:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29] Testing connection with [1]: wss://localhost:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29] Testing connection with [2]: ws://127.0.0.1:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29] Testing connection with [3]: wss://127.0.0.1:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29] Testing connection with [4]: ws://tube.99-prozent-bewegung.net:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29] Testing connection with [5]: wss://tube.99-prozent-bewegung.net:2051 on 2021/09/06 17:51:29
[2021-09-06 17:51:29]  NOT SECURE CONNECTION  Good news, your localhost connects, that means your port [2051] is open ws://localhost:2051
[2021-09-06 17:51:49]    SECURE CONNECTION    CONNECTION FAIL wss://localhost:2051 Connection to tls://localhost:2051 timed out after 20 seconds
[2021-09-06 17:51:49]  NOT SECURE CONNECTION  CONNECTION FAIL ws://127.0.0.1:2051 Connection to tcp://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 17:51:49]    SECURE CONNECTION    CONNECTION FAIL wss://127.0.0.1:2051 Connection to tls://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 17:51:49]  NOT SECURE CONNECTION  CONNECTION FAIL ws://tube.99-prozent-bewegung.net:2051 Connection to tcp://tube.99-prozent-bewegung.net:2051 timed out after 20 seconds
[2021-09-06 17:51:49]    SECURE CONNECTION    CONNECTION FAIL wss://tube.99-prozent-bewegung.net:2051 Connection to tls://tube.99-prozent-bewegung.net:2051 timed out after 20 seconds
[2021-09-06 17:51:49]  We found 0 possible configurations:

**Comment by DanielnetoDotCom** [comment]
first, make sure your site is running under SSL without Cloudflare proxy

FYI, is much easier to make socket work if you use the letsencrypt to generate it

**Comment by peterflitz** [comment]
It is its own rental server, without Cloudflare.
I use Letsencrypt.

But something is wrong, the YPTSocket runs without problems on another server with an identical installation.

Are there any problems with IP6 or is a similar case known?


I bought the Live Users plugin and it only shows live users when YPTSocket is running properly?

**Comment by DanielnetoDotCom** [comment]
Check your configuration.php file and make sure the webSiteRootURL has https on its URL

**Comment by peterflitz** [comment]
It was http ...
Thanks Daniel, overlooked for months.

> Terminal 1:
root@tube:~# sudo php /var/www/html/tube/plugin/YPTSocket/server.php
Starting AVideo Socket server version 2.6 on port 2051
SSL ERROR, we could not access the CRT file /etc/letsencrypt/live/tube.domain.com/@fullchain.pem, try to run this command as root or use sudo
SSL ERROR, we could not access the KEY file /etc/letsencrypt/live/tube.domain.com/@privkey.pem, try to run this command as root or use sudo
Your socket server uses a secure connection
Parameter [local_cert]: /etc/letsencrypt/live/tube.domain.com/@fullchain.pem
Parameter [local_pk]: /etc/letsencrypt/live/tube.domain.com/@privkey.pem
Parameter [allow_self_signed]:
Parameter [verify_peer]:
Parameter [verify_peer_name]:
Parameter [security_level]: 0
root@tube:~#

Unfortunately the timeout is still there:

> Terminal 2.
root@tube:~# sudo php /var/www/html/tube/plugin/YPTSocket/test.php
[2021-09-06 18:57:10] ** Starting socket test **
[2021-09-06 18:57:10] Testing connection with [0]: ws://localhost:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10] Testing connection with [1]: wss://localhost:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10] Testing connection with [2]: ws://127.0.0.1:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10] Testing connection with [3]: wss://127.0.0.1:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10] Testing connection with [4]: ws://tube.domain.com:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10] Testing connection with [5]: wss://tube.domain.com:2051 on 2021/09/06 18:57:10
[2021-09-06 18:57:10]  NOT SECURE CONNECTION  Good news, your localhost connects, that means your port [2051] is open ws://localhost:2051
[2021-09-06 18:57:30]    SECURE CONNECTION    CONNECTION FAIL wss://localhost:2051 Connection to tls://localhost:2051 timed out after 20 seconds
[2021-09-06 18:57:30]  NOT SECURE CONNECTION  CONNECTION FAIL ws://127.0.0.1:2051 Connection to tcp://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 18:57:30]    SECURE CONNECTION    CONNECTION FAIL wss://127.0.0.1:2051 Connection to tls://127.0.0.1:2051 timed out after 20 seconds
[2021-09-06 18:57:30]  NOT SECURE CONNECTION  CONNECTION FAIL ws://tube.domain.com:2051 Connection to tcp://tube.domain.com:2051 timed out after 20 seconds
[2021-09-06 18:57:30]    SECURE CONNECTION    CONNECTION FAIL wss://tube.domain.com:2051 Connection to tls://tube.domain.com:2051 timed out after 20 seconds
[2021-09-06 18:57:30]  We found 0 possible configurations:

**Comment by DanielnetoDotCom** [comment]
Check the SSL files `SSL ERROR, we could not access the CRT file /etc/letsencrypt/live/tube.domain.com/@fullchain.pem,`

make sure you have the correct path on the plugin and restart your socket server.

**Comment by peterflitz** [comment]
I restarted the server.

Now the outputs look different.
The paths to the certificates match.
Live, meet etc everything works correctly and is https.

> Terminal 1:
root@tube:~# sudo php /var/www/html/tube/plugin/YPTSocket/server.php
Starting AVideo Socket server version 2.6 on port 2051
SSL ERROR, we could not access the CRT file /etc/letsencrypt/live/tube.domain.com/@fullchain.pem, try to run this command as root or use sudo
SSL ERROR, we could not access the KEY file /etc/letsencrypt/live/tube.domain.com/@privkey.pem, try to run this command as root or use sudo
Your socket server uses a secure connection
Parameter [local_cert]: /etc/letsencrypt/live/tube.domain.com/@fullchain.pem
Parameter [local_pk]: /etc/letsencrypt/live/tube.domain.com/@privkey.pem
Parameter [allow_self_signed]:
Parameter [verify_peer]:
Parameter [verify_peer_name]:
Parameter [security_level]: 0

> Terminal 2.
root@tube:~# sudo php /var/www/html/tube/plugin/YPTSocket/test.php
[2021-09-06 19:27:13] ** Starting socket test **
[2021-09-06 19:27:13] Testing connection with [0]: ws://localhost:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] Testing connection with [1]: wss://localhost:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] Testing connection with [2]: ws://127.0.0.1:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] Testing connection with [3]: wss://127.0.0.1:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] Testing connection with [4]: ws://tube.domain.com:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] Testing connection with [5]: wss://tube.domain.com:2051 on 2021/09/06 19:27:13
[2021-09-06 19:27:13] ERROR on get connect response [Connection closed before handshake]
[2021-09-06 19:27:13] ERROR on get connect response [Connection to localhost:2051 failed during TLS handshake: Unable to complete TLS handshake: SSL operation failed with code 1. OpenSSL Error messages: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure]
[2021-09-06 19:27:13] ERROR on get connect response [Connection closed before handshake]
[2021-09-06 19:27:13] ERROR on get connect response [Connection to 127.0.0.1:2051 failed during TLS handshake: Unable to complete TLS handshake: SSL operation failed with code 1. OpenSSL Error messages: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure]
[2021-09-06 19:27:13] ERROR on get connect response [Connection closed before handshake]
[2021-09-06 19:27:13] ERROR on get connect response [Connection to tube.domain.com:2051 failed during TLS handshake: Connection lost during TLS handshake]

> Log file
 [06-Sep-2021 19:37:22 Europe/Amsterdam] AVideoLog::ERROR: Could not connect: Connection to www.mysite.com:2051 failed during TLS handshake: Unable to complete TLS handshake: SSL operation failed with code 1. OpenSSL Error messages: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure wss://www.mysite.com:2051?webSocketToken=SlEwYktjM3VBZDBraEpyUFpwdGRvMlBsamhDTXM5NUdjVjRPekVYTHFBc2RNdzFqV2FFY1dMUDRONmkyZFg1cDFzMjEvL2FQeDdjWWU0VCt1aGpMTGZNYW1YN0x4S1lSSmNJYkgrNk0wVWZINjN4cGJJWGNmU1pCRHkwaHhJQVJ2TmV6UmN0ajNkb2Y1UzRXbWw3Z24vTmY1RlBBbXdJaEYzUHVSUGtRbnlaUzAzUU0weWJYMU5ZT3NoZHYydW83M0w5ODIydkNFRmtFaTAxUko0OEgzd3lYUG82UzR5eWhlNXVsOGhVOS9JMzZ4c1RUeVZEc1pPUWpFeTUzZjdPODJ5cWFWWTQzTUFmM3JwSnVFK0dwV3BjbWk4cTg2ZmZWNXR6dCtkdk5yTDExd09EYVNkYlZvcVJtK2xXNzM4R2JjQ2pKR3R3cW5Na014M0tKQTR1d000MUlQaFpVUG1tVkRMNyt0QXp1ck1VSGhpT3gzNmJhTUFUeDhGMmJkcGZnWk92TjJYNFdpdWcva2pqbGNSZ0F0cU5NNTJKeFVQOWxKMjl3S3lGOEdiNVZFN0NTQ20wSFh4d2FFRjE2TExuRXBoUEFHTnZUSzFQdDNDMnJOMFR1cWRGU1RCZFZpbERITUpPb250VmF5a2ZNMkxacks4S1oxN0pMbmRsbE15ckhMWnBpN1hFMHVZUTJxUllGKzhGOGU0UzFpdGZ3Y3VJZk12VHFkMWNZRGlaYlpaL2pvOVJuYzg2OHdQbVpGNEtodFFqUEg4d01oTlpYaStsYkY3L1JVcDh1OWJwZWpCRk5CN2ViNlFlWTI1L2x5eDY4QkEvbGdkZUcvWkV5bjZ5R2F0OVBmdlVJUGtmdzZML0tnd2JjRmc9PQ==&isCommandLine=1 SCRIPT_NAME: /tube/plugin/Live/on_publish_done.php

**Comment by DanielnetoDotCom** [comment]
I still can see an error 

`SSL ERROR, we could not access the CRT file /etc/letsencrypt/live/tube.domain.com/@fullchain.pem`

check your plugin parameters and also make sure the file exists

**Comment by peterflitz** [comment]
483 / 5000
Übersetzungsergebnisse
It's crazy ...

The domain and the live plug-in use these certificates without any problems.
Only the YPTSocket plugin doesn't like this certificate?

The fault must be somewhere else.
Do you have any other idea where the mistake could be?

The difference to the other server (the YPTSocket runs there):
www.domain.com
On the main server (the YPTSocket does not run there):
tube.domain.com

Are there problems with the YPTSocket running there if it is running on a subdomain?

**Comment by DanielnetoDotCom** [comment]
The script is saying the file does not exists

**Comment by peterflitz** [comment]
yes i know but that's the problem.
The certificate is there, it is being used successfully by Plugin Live and from the Website.

Does this log show a hint (termination before handshake)?

>[2021-09-06 19:27:13] ERROR on get connect response [Connection closed before handshake]
[2021-09-06 19:27:13] ERROR on get connect response [Connection to localhost:2051 failed during TLS handshake: Unable to complete TLS handshake: SSL operation failed with code 1. OpenSSL Error messages: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure]

**Comment by DanielnetoDotCom** [comment]
Since the script is saying it cannot access it I can only assume it does not exists or it is a permission issue.

you must resolve this before proceeding

**Comment by peterflitz** [comment]
Ok, found some of the bugs.

I first installed avideo without a certificate.
Then I migrated from another server.

Errors have happened, be it access rights or http entries, strange but not bad.

I checked everything and reinstalled the plugins, now everything works fine.

http in the config file brought me to the possible problem:

> Check your configuration.php file and make sure the webSiteRootURL has https on its URL


Thanks Daniel
